test/example-types/dateRangeFacet.js

```diff
@@ -0,0 +1,87 @@
+let _ = require('lodash/fp')
+let { expect } = require('chai')
+let { getDateIfValid, rollingRangeToDates } = require('../../src/dateUtil')
+let dateRangeFacet = require('../../src/example-types/dateRangeFacet')
+
+let commonFilterParts = {
+  type: 'dateRangeFacet',
+  field: 'test',
+  ranges: [
+    { range: 'allFutureDates', key: 'open' },
+    { range: 'allPastDates', key: 'expired' },
+  ],
+}
+
+let getDatePart = (range, part) =>
+  _.flow(_.get(part), getDateIfValid)(rollingRangeToDates(range, 'UTC'))
+
+describe('validContext', () => {
+  it('should validate a context with all required fields', () => {
+    expect(
+      dateRangeFacet.validContext({
+        field: 'value_field',
+        ranges: [{ key: 'asd', range: 'allFutureDates' }],
+      })
+    ).to.be.true
+  })
+  it('should invalidate a context with no ranges prop', () => {
+    expect(
+      dateRangeFacet.validContext({
+        field: 'value_field',
+      })
+    ).to.be.false
+  })
+  it('should invalidate a context with missing range key', () => {
+    expect(
+      dateRangeFacet.validContext({
+        field: 'value_field',
+        ranges: [{ to: 'now/d' }],
+      })
+    ).to.be.false
+  })
+})
+
+describe('dateRangeFacet/filter', () => {
```

Comment:
Having two tests is probably not needed but having one is, IMO. It checks to make sure the ES JSON is formatted correctly as well as the values match. Since I am testing with the range phrase `allFutureDates` and the resulted date is `"yyyy-MM-ddT00:00:00.000Z"` we are fine as far as the `brittle` part. Time is always the same ...

Reply:
I'm still not convinced this test adds a lot of value, but I think it's okay to keep it.